.github/workflows/precommit.yml

```diff
@@ -240,6 +240,61 @@ jobs:
             .tox/.coverage
             .tox/coverage.xml
 
+  test-no-pandas:
+    name: Test when pandas is not installed
+    needs: build
+    runs-on: ${{ matrix.os.image_name }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - image_name: macos-latest
+            download_name: macos
+        python-version: ["3.9"]
```

Comment:
Do we need all supported python versions?

Reply:
I don't think so? For me this merge gate is there to make sure the our pandas dependency is optional. Unless we require pandas versions specifically in the code, I think this merge gate could test one version only.